website/docs/language/expressions/type-constraints.mdx

```diff
@@ -276,29 +276,19 @@ variable "with_optional_attribute" {
 }
 ```
 
-When evaluating variable values, Terraform will return an error if an object
-attribute specified in the variable type is not present in the given value.
-Marking an attribute as optional changes the behavior in that situation:
-Terraform will instead insert a default value for the missing attribute,
-allowing the receiving module to describe an appropriate fallback behavior.
-
-The `optional` modifier takes one or two arguments. The first argument
-specifies the type of the attribute, and (if given) the second attribute
-defines the default value to use if the attribute is not present. The default
-must be compatible with the attribute type. If no default is specified, a
-`null` value of the appropriate type will be used as the default.
+The `optional` modifier takes one or two arguments.
+- **Type:** (Required) The first argument
+specifies the type of the attribute.
+- **Default:** (Optional) The second argument defines the default value that Terraform should use if the attribute is not present. This must be compatible with the attribute type. If not specified, Terraform uses a `null` value of the appropriate type as the default.
 
-During evaluation, object attribute defaults are applied top-down in nested
-variable types. This means that a given attribute's default value will also
-have any nested default values applied to it later.
+Terraform applies object attribute defaults top-down in nested variable types. This means that Terraform applies the default value you specify in the `optional` modifier first and then later applies any nested default values to that attribute.
 
 ### Example: Nested Structures with Optional Attributes and Defaults
 
-The following configuration defines a variable which describes a number of storage buckets, each of which is used to host a website. This variable type uses several optional attributes, one of which is itself an `object` type with optional attributes and defaults.
+The following example defines a variable for storage buckets that host a website. This variable type uses several optional attributes, including `website`, which is itself an optional `object` type that has optional attributes and defaults.
 
 ```hcl
 terraform {
-  # Optional attributes are currently experimental.
   experiments = [module_variable_optional_attrs]
 }
```

Comment:
This entire block should be removed once the experiment is ended; `experiments = [module_variable_optional_attrs]` will start returning an error once the experiment is concluded.

Reply:
Thanks, fixed!